projects/storefrontlib/src/cms-components/configurator/commons/service/config-utils.service.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { TestBed } from '@angular/core/testing';
+
+import { ConfigUtilsService } from './config-utils.service';
+import { Observable, of } from 'rxjs';
+import {
+  ConfiguratorGroupsService,
+  GenericConfigurator,
+} from '@spartacus/core';
+import { Type } from '@angular/core';
+
+let isGroupVisited: Observable<Boolean> = of(false);
+
+class MockConfiguratorGroupsService {
+  isGroupVisited(): Observable<Boolean> {
+    return isGroupVisited;
+  }
+}
+
+describe('ConfigUtilsService', () => {
+  let classUnderTest: ConfigUtilsService;
+  let configurationGroupService: ConfiguratorGroupsService;
+  const groupId = 'group_01';
+  const owner: GenericConfigurator.Owner = {
+    id: 'testProduct',
+    type: GenericConfigurator.OwnerType.PRODUCT,
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: ConfiguratorGroupsService,
+          useClass: MockConfiguratorGroupsService,
+        },
+      ],
+    });
+
+    classUnderTest = TestBed.inject(ConfigUtilsService);
+    configurationGroupService = TestBed.inject(
+      ConfiguratorGroupsService as Type<ConfiguratorGroupsService>
+    );
+    spyOn(configurationGroupService, 'isGroupVisited').and.returnValue(
+      isGroupVisited
+    );
+  });
+
+  it('should be created', () => {
+    expect(classUnderTest).toBeTruthy();
+  });
+
+  it('should return false because the product has not been added to the cart and the current group was not visited', () => {
+    let result: boolean;
+    const subscription = classUnderTest
+      .isCartEntryOrGroupVisited(owner, groupId)
+      .subscribe((data) => (result = Boolean(data)));
+    expect(result).toBe(false);
+    subscription.unsubscribe();
```

Comment:
I guess shorter and more readable to directly call .unsubscribe after subscribe, then const subscription is not needed

Reply:
good point! I went even further and extracted a getActualResult method where the whole magix takes place. :)